cwms-data-api/src/main/java/cwms/cda/data/dto/stream/StreamLocationNode.java

```diff
@@ -0,0 +1,66 @@
+package cwms.cda.data.dto.stream;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import cwms.cda.api.errors.FieldException;
+import cwms.cda.data.dto.CwmsDTOBase;
+import cwms.cda.data.dto.CwmsId;
+import cwms.cda.formatters.Formats;
+import cwms.cda.formatters.annotations.FormattableWith;
+import cwms.cda.formatters.json.JsonV1;
+
+@FormattableWith(contentType = Formats.JSONV1, formatter = JsonV1.class, aliases = {Formats.DEFAULT, Formats.JSON})
+@JsonDeserialize(builder = StreamLocationNode.Builder.class)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonNaming(PropertyNamingStrategies.KebabCaseStrategy.class)
+public final class StreamLocationNode implements CwmsDTOBase {
+
+    private final CwmsId id; //stream location id
+    private final StreamNode streamNode;
+
+    private StreamLocationNode(Builder builder) {
+        this.id = builder.id;
+        this.streamNode = builder.streamNode;
+    }
+
+    @Override
+    public void validate() throws FieldException {
```

Comment:
same as the other validate comment

Reply:
removed, also updated CwmsDTO to call the validation on required annotations